lib/index.js

```diff
@@ -86,22 +87,27 @@ Optimizely.prototype.initialize = function() {
  */
 
 Optimizely.prototype.track = function(track) {
+  var opts = this.options;
+  var eventProperties = track.properties();
   // Use the new-style API (which is compatible with Classic and X)
   var payload = {
     type: 'event',
     eventName: track.event(),
     tags: track.properties()
   };
 
-  if (payload.tags.revenue) {
+  if (payload.tags.revenue && opts.sendRevenueOnlyForOrderCompleted) {
```

Comment:
On our fork we are applying the logic and the update to eventProperties.revenue before payload is created. payload.tags is just assigned eventProperties in the same way as full stack. 

`  // Format revenue
  if (eventProperties.revenue) eventProperties.revenue = Math.round(eventProperties.revenue * 100);

  // Send to snippet
  // Use the new-style API (which is compatible with Classic and X)
  var payload = {
    type: 'event',
    eventName: eventName,
    tags: eventProperties
  };
  push(payload);

  // Send to SDK
  var optimizelyClientInstance = window.optimizelyClientInstance;
  if (optimizelyClientInstance && optimizelyClientInstance.track) {
    var userId = optimizelyOptions.userId || track.userId() || this.analytics.user().id();
    if (userId) {
      optimizelyClientInstance.track(eventName, userId, attributes, eventProperties);
    }
  }`